File: src/Software Engineer Prepare Kit/22/solution.go
package main

func GetBinarySearchTreeHeight(values []int32, leftChild []int32, rightChild []int32) int32 {
	var bTreeHeight func(node int32) int32
	bTreeHeight = func(node int32) int32 {
		if node == -1 {
			return 0
		}

		leftHeight := bTreeHeight(leftChild[node])
		rightHeight := bTreeHeight(rightChild[node])

		return max(leftHeight, rightHeight) + 1
	}

	return bTreeHeight(0)
}

File: src/Software Engineer Prepare Kit/22/solution.ts
export default function getBinarySearchTreeHeight(values: number[], leftChild: number[], rightChild: number[]): number {
	const bTreeHeight = (node: number): number => {
		if (node === -1) return 0

		const leftHeight = bTreeHeight(leftChild[node]!)
		const rightHeight = bTreeHeight(rightChild[node]!)

		return Math.max(leftHeight, rightHeight) + 1
	}

	return bTreeHeight(0)
}
